python/pyiceberg/table/__init__.py

```diff
@@ -401,9 +423,38 @@ def plan_files(self) -> Iterator[FileScanTask]:
                             metrics_evaluator,
                         )
                         for manifest in manifests
+                        if (manifest.content is None or manifest.content == ManifestContent.DATA)
+                        or (
+                            # Not interested in deletes that are older than the data
+                            manifest.content == ManifestContent.DELETES
+                            and (manifest.sequence_number or INITIAL_SEQUENCE_NUMBER) >= min_sequence_number
+                        )
                     ],
                 )
+            ):
+                if datafile.content is None or datafile.content == DataFileContent.DATA:
+                    data_datafiles.append(datafile)
+                elif datafile.content == DataFileContent.POSITION_DELETES:
+                    deletes_positional.append(datafile)
+                elif datafile.content == DataFileContent.EQUALITY_DELETES:
```

Comment:
Shouldn't this match deletes to data files and then check that there are no equality deletes rather than rejecting any table with an equality delete file?

We only care if we need to apply the delete file.

Reply:
This is fine for now, though. We can move on with this, but it would be nice to separate them out.